### PR TITLE
Fix incorrect subtitles and overloaded member functions in documentation

### LINKED ISSUE #15691: Fixing the display problems of member functions of some classes in documentation
There are some problems in displaying member functions of some classes in our doc, say `Triangulation`, `DofHandler` and `MatrixFree`.

After digging into the header files, I found some **missing bracket pairs** and **incorrect use of doxygen-style brackets** (should be `/** @} */`).

```bash
find ../include/ -type f | xargs grep -Pzo -n '/\*\n +\* @\}\n +\*/\n'
../include/deal.II/grid/tria.h:1:/*
   * @}
   */
# SOME MORE OUTPUT ...
../include/deal.II/dofs/dof_handler.h:1:/*
   * @}
   */
```

Furthermore, the `Triangulation` class contains **8 overloads** of `n_quads` (and some other functions) including the specializations which are not supposed to be parsed by doxygen (the header file uses the `#ifndef DOXYGEN` macro).

I solve this by **deleting the source path** in the `INPUT` field of the generated `options.dox`. But this also eliminate the `source` directory in the [**File List tab**](https://www.dealii.org/current/doxygen/deal.II/files.html).

https://github.com/dealii/dealii/blob/01b805b00b621143e6ca89ff13dfde00807e0e31/doc/doxygen/CMakeLists.txt#L189-L196

I'm wondering if adding the source path into the doxygen input is necessary since functions only defined in the source file should not show up in the doc like [`DataOutBase::get_node_positions`](https://www.dealii.org/current/doxygen/deal.II/namespaceDataOutBase.html#ac50a84c928a2d6bb906815f2ffe255d7).

### PR DESCRIPTION
**Pull request by RichardYCJ**
If merged, this PR will close #15691.

### CODE REVIEW

File: include/deal.II/grid/grid_generator.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ namespace GridGenerator
   /**
    * @name Creating meshes for basic geometries
    */
-  //** @{ */
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h:69]: Ugh. Who thought that was a good idea?

**Review by bangerth — APPROVED** [review]
This looks good to me. What is still missing to remove the WIP label?

**Comment by RichardYCJ** [comment]
> This looks good to me. What is still missing to remove the WIP label?

I'm still thinking of whether we should remove the source path in the `options.dox.in`. I quiet like to dive into source codes in the File List tab but again, we better not to display all the functions in the source file in the doc.

**Comment by bangerth** [comment]
How about leaving this for a separate PR?

Separately, we've already picked up a merge conflict in `tria.h` :-(

**Comment by tamiko** [comment]
@RichardYCJ Would you mind to rebase your branch? (instead of merging `master` into your topics branch).

**Comment by RichardYCJ** [comment]
> Would you mind to rebase your branch? (instead of merging `master` into your topics branch).

Totally ok for me, do I need to rebase and recommit or you can handle that in the review tab?

**Review by masterleinad — APPROVED** [review]

**Comment by masterleinad** [comment]
@RichardYCJ OUr `indent` CI fails with
```
invalid author 'RichardYCJ' without firstname and lastname

hint: for possible solutions, consult the webpage:
      https://github.com/dealii/dealii/wiki/Commit-authorship
```
Would you mind including a proper name in all the commits?

**Comment by RichardYCJ** [comment]
Sorry for the late response, got an emergency to handle. I'm currently working on this, should be done very soon 🙏

**Comment by RichardYCJ** [comment]
Sorry It seems there're still problems within, I would run `check_indentation.sh` myself next time.

**Comment by RichardYCJ** [comment]
should be ok this time 👀 (since I use `clang-format-17`, I have to disable the clangd and do the indentation manually)

**Pull request merged**